Add CONTRIBUTING.md based on our wiki articles

mrmeszaros opened the pull request:
Created an initial version of CONTRIBUTING.md mentioned in #4132 

It contains the contents of the more relevant wiki pages.

Definitely needs some brainstorming, to specify what should go there, and what should be only linked to the wiki.

mrmeszaros committed:
Add CONTRIBUTING.md based on our wiki articles

bilelmoussaoui commented:
The file can go under .github btw ;)

mrmeszaros committed:
Move CONTRIBUTING.md to .github/

mymindstorm commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,66 @@
+
+![inkscape3](https://user-images.githubusercontent.com/7050624/27377856-7afa5020-5676-11e7-93ae-d857b38988ec.png)
+
+## Unthemed icons
```
Quick question: why is the unthemed icons section in CONTRIBUTING.md? I don't see why anyone who is interested in contributing would want this info.

mymindstorm replied:
Also, you might want to change the README to link to CONTRIBUTING.md and we could drop the wiki page since this pull makes it redundant.

Foggalong reviewed (changes requested):
Keep the contribution guidelines for how to submit pull requests and the procedure to follow. In addition to the "Contribution process" add sections for:

1. **New Icon**: link to the design guidelines, reminder to update `data.json`, and inform that user must include Circle & Square and other themes are optional.
2. **Data File**: an explanation of the structure and how to add/change it. This includes when adding an icon
3. **Python**: we use pycodestyle linting, and basic explanation of @bil-elmoussaoui's validation scripts.

The design guidelines for each theme should stay in the wiki, especially considering they're only going to get longer as new themes are added.

The section on unthemed icons is also irrelivent for the contributing guidelines and should remain a wiki age.

mrmeszaros committed:
Remove irrelevant parts

mrmeszaros committed:
Rework `Contribution process` section

Fixed the broken link to commit message guide.

Added citation blocks for important parts to stand out.

mrmeszaros committed:
Fix a typo

palob commented:
Regarding our design guidelines. I'm currently using an Inkscape git build and in the HSL colour notation they changed the ranges for HSL from [0-255,0-255,0-255] to [0-360,0-100,0-100] which afaik is the correct way and how it's done in other graphics applications. This means our baseplate gradient will be 4L instead of 10L as of Inkscape 0.93.

mrmeszaros committed:
Add sections on new icons the data file and python

Added the first draft of the requested sections.

The section on Python needs more info from @bil-elmoussaoui.

mrmeszaros committed:
Add Code Style section

Extended the optimization options, and turned it into a subsection.

[ci skip]

palob commented:
That's gonna be exhaustive, thanks for your efforts!

Foggalong committed:
Cleanup

Structural change to make things read cleaner, removed optimisation settings (probably easier to keep in the wiki), and added external links for more of the definitions and steps.
